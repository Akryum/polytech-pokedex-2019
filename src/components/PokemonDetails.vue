<template>
  <div class="pokemon-details">
    <BaseLoading
      v-if="$apollo.loading"
    />
    <template v-else-if="pokemon">
      <div class="id">
        <span class="symbol">#</span>
        <span class="number">{{ pokemon.id }}</span>
      </div>
      <div class="name">
        {{ pokemon.name }}
      </div>
      <BaseImage
        :src="pokemon.image"
        :alt="`Image of ${pokemon.name}`"
        class="preview"
      />
      <div>
        Habitat: <b>{{ pokemon.habitat }}</b>
      </div>

      <div class="actions">
        <button @click="showEdit = true">
          Edit
        </button>
      </div>
    </template>
    <div
      v-else
      class="empty"
    >
      Pokemon not found
    </div>

    <BaseModal
      v-if="showEdit"
      :title="`Edit Pokemon #${id}`"
      @close="showEdit = false"
    >
      <template #title>
        Edit Pokemon #{{ id }}
      </template>

      <EditPokemon
        :pokemon="pokemon"
        @done="showEdit = false"
      />
    </BaseModal>
  </div>
</template>

<script>
import EditPokemon from './EditPokemon.vue'

import gql from 'graphql-tag'

export default {
  components: {
    EditPokemon
  },

  props: {
    id: {
      type: Number,
      required: true
    }
  },

  data () {
    return {
      showEdit: false
    }
  },

  apollo: {
    pokemon: {
      query: gql`
        query Pokemon ($id: Int!) {
          pokemon (id: $id) {
            id
            name
            image
            habitat
          }
        }
      `,
      variables () {
        return {
          id: this.id
        }
      }
    }
  },

  created () {
    console.log(this.$route.params.id)
  }
}
</script>

<style lang="stylus" scoped>
.pokemon-details
  background $color-grey
  border-radius $br
  margin $padding
  padding $padding

.id
  .symbol
    color $color-grey-dark

.name
  font-size 2em
  font-weight lighter

.preview
  width 192px
  height @width
  >>> .image
    image-rendering pixelated

.actions
  margin-top 12px
</style>
