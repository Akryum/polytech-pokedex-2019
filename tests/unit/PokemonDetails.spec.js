import PokemonDetails from '@/components/PokemonDetails.vue'
import { shallowMount } from '@vue/test-utils'

describe('PokemonDetails', () => {
  const stubs = [
    'BaseImage'
  ]

  const mocks = {
    $route: {
      params: {
        id: 2
      }
    },
    $apollo: {
      loading: false
    },
    pokemon: {
      id: 2,
      name: 'Ivysaur',
      image: 'https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/2.png',
      habitat: 'forest'
    }
  }

  test('snapshot', () => {
    const wrapper = shallowMount(PokemonDetails, {
      propsData: {
        id: 2
      },
      stubs,
      mocks
    })
    expect(wrapper.find('.number').text()).toBe('2')
    expect(wrapper.find('.name').text()).toBe('Ivysaur')
    expect(wrapper).toMatchSnapshot()
  })
})
