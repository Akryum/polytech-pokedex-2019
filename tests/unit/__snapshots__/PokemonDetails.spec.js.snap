// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PokemonDetails snapshot 1`] = `
<div class="pokemon-details">
  <div class="id"><span class="symbol">#</span> <span class="number">2</span></div>
  <div class="name">
    Ivysaur
  </div>
  <baseimage-stub src="https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/2.png" alt="Image of Ivysaur" class="preview"></baseimage-stub>
  <div>
    Habitat: <b>forest</b></div>
  <div class="actions"><button>
      Edit
    </button></div>
  <!---->
</div>
`;
